### PR TITLE
Use LUFA funcs for split_util

### LINKED ISSUE #8592: Allow split keyboards on chips without USB like 328p
Trying to make a split keyboard with a 328p chip. And the software does not compile due to the fact that split_util.c use USB registers. So make this patch for undef USB stuff.

[0001-split-for-chip-without-usb.patch.txt](https://github.com/qmk/qmk_firmware/files/4397818/0001-split-for-chip-without-usb.patch.txt)

### PR DESCRIPTION
**Pull request by zvecr**
## Description

While in the area, i have also cleaned up the logic so that non LUFA avr boards can just implement `is_keyboard_master`.

(cb4a64e6c5013091c8864366320874f9114625c5 was added just to check compile issues, it can be removed if required)

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Partially fixes #8592

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: quantum/split_common/split_util.c [hunk 1]
```diff
@@ -22,30 +26,48 @@
...
 volatile bool isLeftHand = true;
 
-bool waitForUsb(void) {
+#if defined(SPLIT_USB_DETECT)
+#    if defined(PROTOCOL_LUFA)
+static inline bool usbHasActiveConnection(void) { return USB_Device_IsAddressSet(); }
+static inline void usbDisable(void) { USB_Disable(); }
+#    elif defined(PROTOCOL_CHIBIOS)
+static inline bool usbHasActiveConnection(void) { usbGetDriverStateI(&USBD1) == USB_ACTIVE; }
+static inline bool usbDisable() { usbStop(&USBD1) }
```
Comment (fauxpark) [inline quantum/split_common/split_util.c]:
```suggestion
static inline void usbDisable(void) { usbStop(&USBD1); }
```

File: quantum/split_common/split_util.c [hunk 2]
```diff
@@ -22,30 +26,48 @@
...
+#    else
```
Comment (fauxpark) [inline quantum/split_common/split_util.c]:
```suggestion
#    elif defined(PROTOCOL_VUSB)
static inline bool usbHasActiveConnection(void) { return usbConfiguration; }
static inline void usbDisable(void) { usbDeviceDisconnect(); }
#    else
```
This *might* work.

File: quantum/split_common/split_util.c [hunk 3]
```diff
@@ -6,6 +6,10 @@
 #include "transport.h"
 #include "quantum.h"
 
+#ifdef PROTOCOL_LUFA
+#    include <LUFA/Drivers/USB/USB.h>
+#endif
```
Comment (fauxpark) [inline quantum/split_common/split_util.c]:
```suggestion
#endif

#ifdef PROTOCOL_VUSB
#    include "usbdrv.h"
#endif
```

File: quantum/split_common/split_util.c (hunk 1 above)
Comment (zvecr) [reply quantum/split_common/split_util.c]: Nice catch

File: quantum/split_common/split_util.c (hunk 2 above)
Comment (zvecr) [reply quantum/split_common/split_util.c]: I was going to tackle it as part of another round, to get at least one transport tested. But I figure it compiles, so why not.

**Pull request merged**